backend/webapp/src/main/java/org/bonitasoft/web/designer/controller/UploadResourceController.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (C) 2015 Bonitasoft S.A.
+ * Bonitasoft, 32 rue Gustave Eiffel - 38000 Grenoble
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2.0 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.bonitasoft.web.designer.controller;
+
+import javax.inject.Inject;
+
+import org.bonitasoft.web.designer.controller.upload.WebResourceUploader;
+import org.bonitasoft.web.designer.model.page.Page;
+import org.bonitasoft.web.designer.rest.ErrorMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.multipart.MultipartFile;
+
+/**
+ * This controller is used to upload resources and attach them to a component
+ */
+@Controller
+public class UploadResourceController {
+
+    private WebResourceUploader<Page> pageWebResourceUploader;
+
+    @Inject
+    public UploadResourceController(WebResourceUploader<Page> pageWebResourceUploader) {
+        this.pageWebResourceUploader = pageWebResourceUploader;
+    }
+
+    @RequestMapping(value = "/upload/resources/page/{id}", method = RequestMethod.POST)
```

Comment:
IMHO URL should be something like that : `/page/{id}/resources` depending of the json model of the page.
If it's webResources it should be webResources in the url.
Furthermore, if ressource's name is the unique identifier, when can use something like that
PUT /page/{id}/resources/{name}  to create and update resource

What is sure is that should be a "sub controller" has we made for data, so we can attach resource to other components easier